core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -417,6 +423,12 @@ object KafkaConfig {
   val NumRecoveryThreadsPerDataDirDoc = "The number of threads per data directory to be used for log recovery at startup and flushing at shutdown"
   val AutoCreateTopicsEnableDoc = "Enable auto creation of topic on the server"
   val MinInSyncReplicasDoc = "define the minimum number of replicas in ISR needed to satisfy a produce request with acks=all (or -1)"
+  val MessageFormatVersionDoc = "Specify the message format version the broker will use to append messages to the logs. Valid values are \"v0\" and \"v1\" (case sensitive). " +
```

Comment:
Thinking about this again, since it's possible for us to change the message format more than once within the same release and we need to make sure message format is consistent with inter protocol version, it's probably better to just use the values in ApiVersion to specify the message format. We can probably extend each of the case object in ApiVersion to add a magic field to indicate the message version associated with each protocol version. Sorry for going back and forth on this one.